### DISCUSSION TITLE
router.push adds a query when called

### DISCUSSION THREAD

**Discussion by mzaien** [vercel/next.js#27268] [Help] [answered]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

14

### What browser are you using?

Chrome, Safari

### What operating system are you using?

macOS

### How are you deploying your application?

Vercel

### Describe the Bug

When calling router.push(router.query.X) the push goes to the same url but with an addition of a new query 
see below 
![CleanShot 2021-06-30 at 10 26 02](https://user-images.githubusercontent.com/43112535/123919580-bc6fed00-d98d-11eb-86dc-6327a87f3122.gif)

this only happens when you navigate through the web app like in the gif, and if you sign in again it will work so it is a bad User experience  having to sign in twice

some code to make sure it is clear 

```
 const signIn = ({ email, pass }) => {
    auth
      .signinWithEmail(email, pass)
      .then(() => {
         router.push(router.query.url);
      })
```

### Expected Behavior

it should go the the /router.query.url directly

### To Reproduce

You can try it here 
https://drive-web-nyqerhgjh-mzaien.vercel.app/SignIn/?url=/
you can sign in with google it gives the same effect

**Comment by mzaien** [comment]
i have used `window.location.herf` && `window.location.replace` and all of them work greatly

**Comment by ijjk** [comment]
Can you provide a minimal reproduction/the related repo for where you are seeing this behavior, calling `router.push('/')` will not add a query to the URL so it sounds like a different value/a different `router.push` is updating the URL.

**Comment by mzaien** [comment]
Sorry for not making the steps more clear

1. visit https://drive-web-nyqerhgjh-mzaien.vercel.app/SignIn/?url=/
2. sign in with google or the email in the gif 
3. you will be redirected to the home page 
4. click on any majors button 
5. logout 
6. sign in again and watch the link changing

**Comment by elrafa-sg** [comment]
My guess is that you're calling router.push('/') inside a form. 
If you're doing this, you'll need to prevent default action of form.

Try something like this:

```
function someFunction(e) {
  e.preventDefault()
  router.push('/')
}

<form onSubmit={(e) => someFunction()>
     // inputs e etc here
</form>
```

**Comment by mzaien** [comment, marked as answer]
Okay it is fixed with a work around, i was passing the url query as `router.asPath` but know i made a constant 
`const pastRouter = router?.asPath` and it works flawlessly  i think there is something being added while the user is using the site 

Thanks @ijjk @elrafa-sg 

i'll open source the project soon and if it you need any investigations i'm happy to help
